component/module/module.go

```diff
@@ -38,10 +39,16 @@ func NewModuleComponent(o component.Options) (*ModuleComponent, error) {
 
 // LoadFlowContent loads the flow controller with the current component content. It
 // will set the component health in addition to return the error so that the consumer
-// can rely on either or both.
+// can rely on either or both. If the content is the same as the last time it was
+// successfully loaded, it will not be reloaded.
 func (c *ModuleComponent) LoadFlowContent(args map[string]any, contentValue string) error {
+	if contentValue == c.getLatestContent() {
+		return nil
+	}
+
 	err := c.mod.LoadConfig([]byte(contentValue), args)
 	if err != nil {
+		c.setLatestContent("")
```

Comment:
reset the latestContent so if content fails to load we are able to retry it

Reply:
Maybe add a comment for this?

I'd personally find this surprising, since based on Flow's guarantees, the last config continues to be used. 

Since you only call `setLatestContent` when there's not an error, I'm not sure this is necessary; you can still try with content that hasn't been used before.

Reply:
o ya actually that makes sense. Let me not set it in this case so the latest config is still correctly stored here.